### DISCUSSION TITLE
Convert pdf file to png fail

### DISCUSSION THREAD

**Discussion by sebUbi** [ImageMagick/ImageMagick#1923] [General]
Hi everybody,

I need some help with converting a pdf file to png.

For most of the files, it works right, but with some files, the pngs generated are just blank (ex : https://www.scoplan.com/uploads/5534-000.png)

It happends only on some files, and on my production and preproduction server (it works on my local server).

Did someone already have this problem ?
 
The code I use is : 

`$im = new \Imagick();
                $im->pingImage('tmp/'.$file->getId().'.'.$extension);
                $numberPages = $im->getNumberImages();
                for ($i=0; $i < $numberPages; $i++) {
                    $im->setResolution(72,72);
                    $im->readimage("tmp/".$file->getId().".".$extension."[".$i."]");
                    $im->setImageFormat('png');    
                    $im->writeImage("tmp/".$file->getId()."-".$i.".png");
                    $im->clear();
                }
                $im->destroy();`

**Comment by snibgo** [comment]
What version of IM? What version of Ghostscript?

Please link to a sample PDF that has the problem, so we can try to reproduce it.

**Comment by sebUbi** [comment]
I cannot give you examples of document as there are private files.

But talking about version : 
local imagick : ImageMagick 6.9.10-23
local ghostscript : GPL Ghostscript 9.27
server imagick : ImageMagick 6.8.9-9
server ghostscript : GPL Ghostscript 9.06

could it be a version problem ?

**Comment by snibgo** [comment]
As it works on your local machine, with more recent software, I suspect it is a version problem.

I suggest you also check the PDF is visible with Adobe Acrobat Reader.

**Comment by sebUbi** [comment]
thank you for your answer. Everything is fine in Adobe acrobat reader, the only problem I have is when I try to convert it to png format.

I'll try to upgrade my imagick first (hope this will not break my application) and then try to upgrade ghostscript
